mesonbuild/interpreter.py

```diff
@@ -1908,25 +1907,10 @@ def __init__(self, build: 'build.Build', interpreter: 'Interpreter'):
                              })
 
     def _find_source_script(self, prog: T.Union[str, mesonlib.File, ExecutableHolder], args):
+        
         if isinstance(prog, (ExecutableHolder, ExternalProgramHolder)):
             return self.interpreter.backend.get_executable_serialisation([unholder(prog)] + args)
-        # Prefer scripts in the current source directory
-        search_dir = os.path.join(self.interpreter.environment.source_dir,
-                                  self.interpreter.subdir)
-        key = (prog, search_dir)
-        if key in self._found_source_scripts:
-            found = self._found_source_scripts[key]
-        elif isinstance(prog, mesonlib.File):
-            prog = prog.rel_to_builddir(self.interpreter.environment.source_dir)
-            found = dependencies.ExternalProgram(prog, search_dir=self.interpreter.environment.build_dir)
-        else:
-            found = dependencies.ExternalProgram(prog, search_dir=search_dir)
-
-        if found.found():
-            self._found_source_scripts[key] = found
-        else:
-            m = 'Script or command {!r} not found or not executable'
-            raise InterpreterException(m.format(prog))
+        found = self.interpreter.func_find_program({}, prog, {}).held_object
```

Comment:
nitpicking: I think you should use find_program_impl() instead.

Reply:
Note, a difference if you use find_program_impl() is it's silent by default. I'm not sure we want to log the this program lookup, it could be spamming a bit too much IMHO.